indico/modules/events/api.py

```diff
@@ -551,13 +552,26 @@ def _build_event_api_data(self, event):
             'modificationDate': self._serialize_date(event.as_legacy.getModificationDate()),
             'creationDate': self._serialize_date(event.created_dt),
             'creator': self._serialize_person(event.creator, person_type='Avatar', can_manage=can_manage),
-            'hasAnyProtection': event.as_legacy.hasAnyProtection(),
+            'hasAnyProtection': event.effective_protection_mode != ProtectionMode.public,
             'roomMapURL': event.room.map_url if event.room else None,
-            'visibility': Conversion.visibility(event.as_legacy),
             'folders': build_folders_api_data(event),
             'chairs': self._serialize_persons(event.person_links, person_type='ConferenceChair', can_manage=can_manage),
             'material': build_material_legacy_api_data(event) + [build_note_legacy_api_data(event.note)]
         })
+
+        event_category_path = event.category.chain
+        visibility = {'id': '', 'name': 'Everywhere'}
+        if event.visibility == 0:
+            visibility['name'] = 'Nowhere'
+        elif event.visibility:
```

Comment:
```
if event.visibility is None:
    pass  # keep default
elif event.visibility == 0:
    visibility['name'] = 'Nowhere'
else:
    ... 
```

Just to make it clear that the `== 0` is needed; otherwise someone might optimize it away at some point in the future ;)